bundles/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/internal/handler/PLCBridgeHandler.java

```diff
@@ -216,38 +240,53 @@ public void initialize() {
-        configured = configured && (config.get().getRemoteTSAP() != null);
-
-        if (configured) {
-            if (client == null) {
-                client = new PLCLogoClient();
+        config = getConfigAs(PLCLogoBridgeConfiguration.class);
+
+        var client = this.client;
+        if (client == null) {
+            client = new PLCLogoClient();
+            if (!client.isConnected()) {
+                final var localTSAP = config.getLocalTSAP();
+                final var remoteTSAP = config.getRemoteTSAP();
+                if ((localTSAP != null) && (remoteTSAP != null)) {
+                    final var result = client.Connect(config.getAddress(), localTSAP, remoteTSAP);
+                    if (result != 0) {
+                        String message = String.format("Can not initialize LOGO!. %s.", S7Client.ErrorText(result));
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+                    }
+                } else {
+                    String message = "Can not initialize LOGO!. Please, check ip address / TSAP settings.";
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message);
+                }
             }
-            configured = connect();
-        } else {
-            String message = "Can not initialize LOGO!. Please, check ip address / TSAP settings.";
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message);
         }
 
-        if (configured) {
-            String host = config.get().getAddress();
+        if (client.isConnected()) {
+            final var host = config.getAddress();
+
+            var readerJob = this.readerJob;
             if (readerJob == null) {
-                Integer interval = config.get().getRefreshRate();
+                Integer interval = config.getRefreshRate();
                 logger.info("Creating new reader job for {} with interval {} ms.", host, interval);
                 readerJob = scheduler.scheduleWithFixedDelay(dataReader, 100, interval, TimeUnit.MILLISECONDS);
+                this.readerJob = readerJob;
             }
+
+            var rtcJob = this.rtcJob;
             if (rtcJob == null) {
                 logger.info("Creating new RTC job for {} with interval 1 s.", host);
                 rtcJob = scheduler.scheduleAtFixedRate(rtcReader, 100, 1000, TimeUnit.MILLISECONDS);
+                this.rtcJob = rtcJob;
             }
 
+            this.client = client;
             updateStatus(ThingStatus.ONLINE);
         } else {
-            String message = "Can not initialize LOGO!. Please, check network connection.";
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
-            client = null;
+            if (ThingStatus.OFFLINE != getThing().getStatus()) {
+                String message = "Can not initialize LOGO!. Please, check network connection.";
```

Comment:
Casn you consider tto use i18n keys to make it possible to add translations? (see others too)

Reply:
Yes. Will raise next request.